src/libcore/slice/mod.rs

```diff
@@ -3288,6 +3288,38 @@ impl<'a, T> IterMut<'a, T> {
     pub fn into_slice(self) -> &'a mut [T] {
         unsafe { from_raw_parts_mut(self.ptr, len!(self)) }
     }
+
+    /// Views the underlying data as a subslice of the original data.
+    ///
+    /// To avoid creating `&mut` references that alias, this has a
+    /// borrowed lifetime from the iterator.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// # #![feature(slice_iter_mut_as_slice)]
+    /// // First, we declare a type which has `iter_mut` method to get the `IterMut`
```

Comment:
When you say "declare a type", this sounds like a struct definition or a type alias, not a `let` binding.

Reply:
In my defense, I mostly copied this from `Iter::as_slice`. :sweat_smile: